2021-01-19 20:38:22,480 INFO {thd:140138064389952} [arg_parse.py => 72] : use dataset MNIST and model LeNet5
2021-01-19 20:38:36,412 INFO {thd:140135446890240} [trainer.py => 144] : training_set_size is 49995
2021-01-19 20:38:36,412 INFO {thd:140135446890240} [trainer.py => 148] : use device cuda:0
2021-01-19 20:38:36,412 INFO {thd:140138064389952} [simulator.py => 52] : begin training
2021-01-19 20:38:38,957 INFO {thd:140135446890240} [trainer.py => 359] : begin training, hyper_parameter is epochs:1 batch_size:64 learning_rate:0.01 weight_decay:1 optimizer:<class 'torch.optim.sgd.SGD'>, optimizer is SGD (
Parameter Group 0
    dampening: 0
    lr: 0.01
    momentum: 0.9
    nesterov: False
    weight_decay: 2.0002000200020003e-05
) ,lr_scheduler is <torch.optim.lr_scheduler.ReduceLROnPlateau object at 0x7f73f25f7ed0>, model: QuantedModel, loss_fun: NLLLoss(), parameter number is 61706
2021-01-19 20:38:39,377 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 0, learning rate: [0.01], batch training loss: 2.306519031524658
2021-01-19 20:38:41,379 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 78, learning rate: [0.01], batch training loss: 1.9225573539733887
2021-01-19 20:38:43,398 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 156, learning rate: [0.01], batch training loss: 0.5016201138496399
2021-01-19 20:38:45,398 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 234, learning rate: [0.01], batch training loss: 0.5053999423980713
2021-01-19 20:38:47,424 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 312, learning rate: [0.01], batch training loss: 0.25723421573638916
2021-01-19 20:38:49,418 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 390, learning rate: [0.01], batch training loss: 0.22590187191963196
2021-01-19 20:38:51,440 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 468, learning rate: [0.01], batch training loss: 0.2714819312095642
2021-01-19 20:38:53,308 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 546, learning rate: [0.01], batch training loss: 0.15014544129371643
2021-01-19 20:38:55,297 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 624, learning rate: [0.01], batch training loss: 0.09085780382156372
2021-01-19 20:38:57,311 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 702, learning rate: [0.01], batch training loss: 0.28776001930236816
2021-01-19 20:38:59,269 INFO {thd:140135446890240} [trainer.py => 375] : epoch: 1, batch: 780, learning rate: [0.01], batch training loss: 0.14432217180728912
2021-01-19 20:38:59,410 INFO {thd:140135446890240} [trainer.py => 397] : epoch: 1, training loss: 0.45085881255199683
2021-01-19 20:39:00,682 INFO {thd:140135446890240} [inference.py => 62] : use device cuda
2021-01-19 20:39:04,615 INFO {thd:140135446890240} [trainer.py => 426] : epoch: 1, learning_rate: [0.01], validation loss: 0.10859853029251099, accuracy = 0.9655172413793104
2021-01-19 20:39:05,769 INFO {thd:140135446890240} [inference.py => 62] : use device cuda
2021-01-19 20:39:09,710 INFO {thd:140135446890240} [trainer.py => 475] : epoch: 1, learning_rate: [0.01], test loss: 0.0956864133477211, accuracy = 0.9683
2021-01-19 20:39:09,817 DEBUG {thd:140135446890240} [trainer.py => 292] : call ReduceLROnPlateau for total loss 0.5594573428445078
2021-01-19 20:39:10,950 INFO {thd:140135446890240} [inference.py => 62] : use device cuda
2021-01-19 20:39:10,950 ERROR {thd:140135446890240} [executor_pool.py => 37] : catch exception:'Conv2d' object has no attribute '_modules'
2021-01-19 20:39:10,951 ERROR {thd:140135446890240} [executor_pool.py => 38] : traceback:Traceback (most recent call last):
  File "/home/cicie/.local/lib/python3.7/site-packages/cyy_naive_lib-0.1-py3.7.egg/cyy_naive_lib/data_structure/executor_pool.py", line 35, in process
    fn(*args, **kwargs)
  File "/home/cicie/distributed_learning_simulator/fed_quant_worker.py", line 52, in train
    res = inferencer.inference()
  File "/home/cicie/.local/lib/python3.7/site-packages/cyy_naive_pytorch_lib-0.1-py3.7.egg/cyy_naive_pytorch_lib/inference.py", line 127, in inference
    loss = super().inference(**kwargs)
  File "/home/cicie/.local/lib/python3.7/site-packages/cyy_naive_pytorch_lib-0.1-py3.7.egg/cyy_naive_pytorch_lib/inference.py", line 63, in inference
    self.__model_with_loss.set_model_mode(self.__phase)
  File "/home/cicie/.local/lib/python3.7/site-packages/cyy_naive_pytorch_lib-0.1-py3.7.egg/cyy_naive_pytorch_lib/model_loss.py", line 53, in set_model_mode
    self.model.eval()
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1293, in eval
    return self.train(False)
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1277, in train
    module.train(mode)
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1277, in train
    module.train(mode)
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1277, in train
    module.train(mode)
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1276, in train
    for module in self.children():
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1172, in children
    for name, module in self.named_children():
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1190, in named_children
    for name, module in self._modules.items():
  File "/home/cicie/miniconda/envs/fedml/lib/python3.7/site-packages/torch/nn/modules/module.py", line 779, in __getattr__
    type(self).__name__, name))
torch.nn.modules.module.ModuleAttributeError: 'Conv2d' object has no attribute '_modules'

2021-01-19 20:39:10,952 INFO {thd:140138064389952} [simulator.py => 54] : end training
